### ISSUE TITLE
Ability to parse string to integer

### ISSUE THREAD

**Issue by austince** [mikefarah/yq#1664]
**Please describe your feature request.**

Sometimes there are integers encoded as strings (because, sigh) and it would be nice to be able to parse them dynamically for things like sorting.

**Describe the solution you'd like**
If we have data1.yml like:
(please keep to around 10 lines )

```yaml
- version: "10"
- version: "9"
- version: "11"
```

And we run a command:

```bash
yq 'sort_by(.version | parseint)'
```

it could output

```yaml
- version: "9"
- version: "10"
- version: "11"
```

**Describe alternatives you've considered**

I tried adding a `0` to try to force integer evaluation to no avail. 

**Additional context**
Generally being able to change the data type of a field while processing would be useful. Is this already available in some way I haven't found?

**Comment by austince** [comment]
Ah, found something along these lines is possible with tag manipulation and `map_values`.

````shell
map_values(. | .version type = "!!int")
```

A proper `parseint` func still might be clearer and handle errors better? Unsure.

**Comment by mikefarah** [comment]
parse_int and other scalars make sense I think,  I'd take a PR for this.

In the meantime, you can actually use the yaml decode operator which will automatically detect the type:

```
yq '"3" | @yamld' -n
3
yq '"3" | @yamld | tag' -n
!!int
```

**Comment by venkatesh2090** [comment]
Hi, is this a duplicate of #71 as well?

**Comment by mikefarah** [comment]
Yep 👍🏼  This is fixed in the 4.40 pre-release here: https://github.com/mikefarah/yq/releases/tag/v4.40.1 - will be looking to make an official 4.41 release soon

**Comment by mikefarah** [comment]
Fixed in 4.40.3
